llvm/test/CodeGen/RISCV/wide-scalar-shift-by-byte-multiple-legalization.ll

```diff
@@ -723,98 +723,117 @@ define void @lshr_16bytes(ptr %src.ptr, ptr %byteOff.ptr, ptr %dst) nounwind {
 ;
 ; RV32I-LABEL: lshr_16bytes:
 ; RV32I:       # %bb.0:
-; RV32I-NEXT:    addi sp, sp, -48
-; RV32I-NEXT:    sw s0, 44(sp) # 4-byte Folded Spill
-; RV32I-NEXT:    sw s1, 40(sp) # 4-byte Folded Spill
-; RV32I-NEXT:    sw s2, 36(sp) # 4-byte Folded Spill
-; RV32I-NEXT:    lbu a3, 0(a0)
-; RV32I-NEXT:    lbu a4, 1(a0)
+; RV32I-NEXT:    addi sp, sp, -32
+; RV32I-NEXT:    lbu a3, 1(a0)
```

Comment:
This looks worse?

Reply:
the same as for aarch64, plus it seems that the load/store alignment used in the test, which is 1, does not mix well with the legalized code.

Reply:
I think it's strange this test is using an unaligned load in the first place, but this shouldn't be regressing the load that's unrelated to the use shift

Reply:
I checked it again, and the reason is the same as in for aarch64. We need a load+shift here.

Reply:
I will add some tests where the shift amount is known to be a multiple of 4 bytes, so 1 load with an offset is enough to do the shift.

Reply:
The test uses unaligned load/store for the shiftee (which lowers to LBUs), but we use aligned load/store for stack operations, which lowers to LWs. This means that the registers used for loading the shiftee cannot be used directly for storing the shiftee on the stack, and the registers used for loading the shifted value from the stack cannot be used directly to store the result.

LBU: loads a 8-bit value from memory, then sign-extends to 32-bits before storing in the destination register
LW: loads a 32-bit value from memory into the destination register